### DISCUSSION TITLE
Getter를 지양해야 하는 이유에 대해 말해주세요

### DISCUSSION THREAD

**Discussion by J-I-H-O** [2024-woowacourse-study/level-interview#16] [OOP]
글쎄요 멀까요?

**Comment by 3Juhwan** [comment]
1. 캡슐화 원칙 위배: getter()는 객체의 내부 상태를 외부에서 직접 접근할 수 있게 만듭니다. 이는 캡슐화 원칙을 위배하며, 객체의 내부 구현 세부사항이 외부에 노출되어 예상치 못한 변경이 일어나고 유지보수를 어렵게 만들 수 있습니다.

2. 불변 객체와 상태 변경 방지: getter()를 통해 객체의 상태를 외부에서 읽을 수 있으면 불변 객체를 유지하기 어렵습니다. 

3. 의존성 증가: 외부에서 getter()에 의존하는 코드가 많아집니다. 해당 클래스에 변경이 발생하면 다른 코드에 미치는 영향을 증가합니다.

getter() 대신에 객체의 상태를 변경하는 메서드를 제공하고, 필요한 경우 객체의 상태를 읽을 수 있는 메서드를 명확하게 정의하는 것이 좋습니다.

**Comment by J-I-H-O** [comment]
getter 메서드를 사용하게 되면 캡슐화가 깨지게 됩니다. 객체의 내부 상태를 외부에서 직접 접근할 수 있게 하는 것이기 때문입니다. 또한 객체의 내부 구현에 대한 의존성이 높아지게 됩니다.

(추가로 볼 것) 디미터의 법칙
